src/main/java/it/pagopa/pn/deliverypush/service/impl/TimeLineServiceImpl.java

```diff
@@ -211,4 +218,80 @@ public boolean isPresentTimeLineElement(String iun, Integer recIndex, TimelineEv
         return this.timelineDao.getTimelineElement(iun, timelineEventId.buildEventId(eventId)).isPresent();
     }
 
+    public void enrichTimelineElementWithConfidentialInformation(TimelineElementDetailsInt details,
+                                                                 ConfidentialTimelineElementDtoInt confidentialDto) {
+        if( details instanceof CourtesyAddressRelatedTimelineElement){
+            CourtesyDigitalAddressInt address = ((CourtesyAddressRelatedTimelineElement) details).getDigitalAddress();
+
+            if (address == null)
+            {
+                address = CourtesyDigitalAddressInt.builder().build();
+            }
+
+            address = address.toBuilder().address(confidentialDto.getDigitalAddress()).build();
+            ((CourtesyAddressRelatedTimelineElement) details).setDigitalAddress(address);
+        }
+
+        if( details instanceof DigitalAddressRelatedTimelineElement && confidentialDto.getDigitalAddress() != null){
+
+            LegalDigitalAddressInt address = ((DigitalAddressRelatedTimelineElement) details).getDigitalAddress();
+
+            if (address == null)
+            {
+                address = LegalDigitalAddressInt.builder().build();
+            }
+
+            address = address.toBuilder().address(confidentialDto.getDigitalAddress()).build();
+            ((DigitalAddressRelatedTimelineElement) details).setDigitalAddress(address);
+        }
+
+        if( details instanceof PhysicalAddressRelatedTimelineElement){
+            PhysicalAddressInt physicalAddress = ((PhysicalAddressRelatedTimelineElement) details).getPhysicalAddress();
+
+            if (physicalAddress == null)
+            {
+                physicalAddress = PhysicalAddressInt.builder().build();
+            }
+
+            physicalAddress = physicalAddress.toBuilder()
+                    .at(confidentialDto.getPhysicalAddress().getAt())
+                    .address(confidentialDto.getPhysicalAddress().getAddress())
+                    .municipality(confidentialDto.getPhysicalAddress().getMunicipality())
+                    .province(confidentialDto.getPhysicalAddress().getProvince())
+                    .addressDetails(confidentialDto.getPhysicalAddress().getAddressDetails())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 223-233, lines 235-246, lines 248-267, lines 269-293 into separate methods.